library/cipher.c

```diff
@@ -413,6 +414,10 @@ int mbedtls_cipher_setkey(mbedtls_cipher_context_t *ctx,
     }
 
     return MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA;
+#else
+    return mbedtls_cipher_get_base(ctx->cipher_info)->setkey_enc_func(ctx->cipher_ctx, key,
+                                                                      ctx->key_bitlen);
+#endif
```

Comment:
Note that `mbedtls_cipher_setkey` is a public API, the user could call it with `mode = MBEDTLS_DECRYPT` (e.g. for `MBEDTLS_CIPHER_AES_128_ECB`). No functions would tell the user "The current build doesn't support decryption of AES_128_ECB" until they call `mbedtls_cipher_update` and got a segmentation fault because there is no decryption functions.

Here the question comes: even we only expected `CIPHER_ONLY` being auto-enabled from PSA configuration, is it convincing to say that the user are only expected to use PSA API in that case?
I think at least we should return some errors from legacy API when users are requesting unsupported operations if `CIPHER_ONLY` is enabled.